### PR TITLE
kmcprintf OK

### PR DESCRIPTION
**Pull request by hensldm**
Matches kmcprintf for libgultra_d 2.0J.

### CODE REVIEW

File: src/error/kmcprintf.c [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+        q = xbuf;
+        while (q != p) {
+            rmonPutchar(*q++);
+        }
+    }
+    return (void*)1;
+}
+
+char NULSTR[] = { '\0' };
```
Comment (AngheloAlf) [inline src/error/kmcprintf.c:84]: Is this required? Can it be this instead?
```suggestion
char NULSTR[] = "";
```

File: src/error/kmcprintf.c [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            }
+            p = xbuf;
+        }
+    }
+    if (p != xbuf) {
+        rmonPutchar((p - xbuf) - 1);
+
+        q = xbuf;
+        while (q != p) {
+            rmonPutchar(*q++);
```
Comment (AngheloAlf) [inline src/error/kmcprintf.c:78]: Could this be written as a for?

File: src/error/kmcprintf.c [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    "alSeqNextEvent: Read past end of sequence",
+    "osAiSetNextBuffer: DMA buffer location may cause audio clicks (0x%x)",
+    "_loadOutputBuffer: Modulated delay greater than total delay by %d samples",
+    "osViExtendVStart: VI Manager not yet begun by osCreateViManager",
+    "osViExtendVStart: value not in range [0-48] %d",
+    NULSTR,
+};
+
+static void kmcErrorHandler(s16 code, s16 numArgs, ...);
+OSErrorHandler __kmcErrorHandler = kmcErrorHandler;
```
Comment (AngheloAlf) [inline src/error/kmcprintf.c:227]: I suppose this has to be here and can't be below, right?

File: src/error/kmcprintf.c (hunk 2 above)
Comment (hensldm) [reply src/error/kmcprintf.c:78]: Not really. It really wants `rmonPutchar(*q++)` in order to match.
```c
            for(q = xbuf; q != p;) {
                rmonPutchar(*q++);
            }
```
works but the while seems better at that point.

File: src/error/kmcprintf.c (hunk 3 above)
Comment (hensldm) [reply src/error/kmcprintf.c:227]: Correct, the data symbol comes before the text symbol of the function.

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**